hash-tables/overview-of-hash-tables.md

```diff
@@ -0,0 +1,153 @@
+# Overview of Hash Tables
+
+## Learning Goals
+
+- Define hash table
+- Define hash function
+- Describe how hash tables resolve collisions
+
+## Introduction
+
+Python dictionaries are fascinating! They store key-value pairs, and, as long as we have the key, getting the value has a time complexity of O(1).
```

Comment:
cut the reason here. instead develop an example with a stronger parallel between lists and dicts.
```suggestion
Python dictionaries are fascinating!
```